lib/options.nix

```diff
@@ -134,6 +134,12 @@ rec {
       throw "The unique option `${showOption loc}' is defined multiple times. Definition values:${showDefs defs}"
     else (head defs).value;
 
+  mergeUniqueOption = { message }: loc: defs:
```

Comment:
`mergeOneOption` is a public `lib` function with a bad name.

I'm not making the message optional, because it seems like a good idea to require it.

Reply:
Same as below, I think we could at least do `mergeOneOption = mergeUniqueOption { message = ""; }` to avoid duplicated code.

Also not sure I agree about requiring a more detailed error message. Why here in particular?

Reply:
For example what should the message be when merging several values of `package` type? I think the base message is self-explanatory

Reply:
Fair enough, perhaps many options are quite obvious and don't need it.